### DISCUSSION TITLE
Possible performance tuning guidance?

### DISCUSSION THREAD

**Discussion by thecatontheflat** [saul/demofile#256] [Questions] [answered]
Just a hypothetical question. 

Let's say I have built a parser based on the demofile. The parser only listens to certain events (e.g. weapon_fire) and nothing else. My assumption is that there is a bunch of other stuff happening in the demofile that could be "discarded" for the specific use-case, so that my parser could gain some performance boost.

I guess that the demofile logic is tightly coupled, so that it won't be possible to achieve what I want. However, If my assumption is correct, could you please point me in the right direction (e.g. which logical parts should I look into that could be "safely" disabled)?

Cheers

**Comment by saul** [comment]
Unfortunately there is no way (at the moment) to disable particular features to improve performance. It depends exactly what you care about - you likely still want entities to be parsed? If not then you can likely significantly reduce runtime.

**Comment by thecatontheflat** [comment]
Thanks!

**Comment by Z-a-r-a-k-i** [comment]
Could you provide us with an example on how to improve performance if for instance I just want to parse the player list on start of first and second side and retrieve header informations ?

**Comment by Z-a-r-a-k-i** [comment]
Btw thank you for this project @saul it's really doing an amazing job, but I have a parsing time of sometime 10 to 20 seconds just to display the player list in my browser :'(..

**Comment by saul** [comment]
This would require a change to not parse entity updates. I imagine that constitutes the vast majority of the runtime.

**Comment by Z-a-r-a-k-i** [comment]
Do you think that there is a simple hackish way to do it before working on some PR material ?

**Comment by Z-a-r-a-k-i** [comment]
Maybe a function parseNextTick would allow us to be able to parse only the beginning of the demo file: https://github.com/StatsHelix/demoinfo/blob/51044a43590a30cc2796c392f24e3b78a31660fe/DemoInfo/DemoParser.cs#L559 (ironically in c# it's not needed because the parser takes about 1 or 2 sec to parse big demo file)

**Comment by saul** [comment, marked as answer]
You can just call `demoFile#cancel` as soon as you've finished parsing the bit you want.

**Comment by Z-a-r-a-k-i** [comment]
> You can just call `demoFile#cancel` as soon as you've finished parsing the bit you want.

Nice man, it did the trick, thank you very much !!
